src/common/src/array/arrow/arrow_iceberg.rs

```diff
@@ -168,6 +170,13 @@ impl ToArrow for IcebergArrowConvert {
             .map_err(ArrayError::from_arrow)?;
         Ok(Arc::new(array) as ArrayRef)
     }
+
+    fn interval_to_arrow(
```

Comment:
Out of curiosity, why is it unnecessary to have a counterpart in the `FromArrow` implementation below to convert from varchar back to interval?

Reply:
Because once we convert an interval type to varchar type, we can't tell what this varchar arrow array solely should be converted to. Currently, we will cast this varchar array on the operator (using a project to cast the varchar back to interval) level.

Reply:
> we can't tell what this varchar arrow array solely should be converted to

Sounds like the extension type annotation can help. But we do not have to go this way if that is a big refactor and has little other benefits.